core/src/main/scala/org/ensime/core/Analyzer.scala

```diff
@@ -267,6 +269,22 @@ class Analyzer(
         val sourceFile = createSourceFile(fileInfo)
         StructureView(scalaCompiler.askStructure(sourceFile))
       }
+    case ASTAtPointReq(file, offset, printTypes,
+      printIds, printOwners, printKinds,
+      printMirrors, printPositions) =>
+      sender ! withExisting(file) {
+        val p = pos(file, offset)
+        scalaCompiler.askLoadedTyped(p.source)
+        val ast = scalaCompiler.locateTree(p)
+        //This is really weird, but if I supply all 6 flags here, compilation fails with type error as
+        //compiler fails to find implicit conversion Option[Boolean] => BoooleanFlag
+        //val rawAst = scalaCompiler.showRaw(ast, printTypes, printIds, printOwners,
+        //                                   printKinds, printMirrors, printPositions)
+        //val rawAst = scalaCompiler.showRaw(ast, true, true, true, true, true, true) (this fails to compile too)
+        val rawAst = scalaCompiler.showRaw(ast, printTypes, printIds, printOwners,
```

Comment:
yeah, I've noticed that as well when working on compiler plugins. also it changes across scala versions, there is something really funky going on. Did you notice that named parameters fail too? I think it has something to do with crazy amountsn of method overloading inside scalac.